### PR TITLE
Added Class Comments for all classes except the ServiceProvider in th…

### LINKED ISSUE #230: Missing Class Comments
There are missing Class Comments in our project.
Especially in our Tests, but Class Comments are not necessary there, we should have a discussion about that

### PR DESCRIPTION
**Pull request by timgarrels**
…e UI and Test Packages

This fixes #230

### CODE REVIEW

File: packages/IMAPClient-Core.package/ICAuthenticationInfo.class/README.md [hunk 1]
```diff
@@ -1 +1,2 @@
-An ICAuthenticationInfo is a wrapper for a username-password-combination.
\ No newline at end of file
+An ICAuthenticationInfo is a wrapper for a username-password-combination.
+It also secures that the strings are IMAP Conform Data
```
Comment (EightSQ) [inline packages/IMAPClient-Core.package/ICAuthenticationInfo.class/README.md]: The right wording here would be "It also _ensures_".

File: packages/IMAPClient-Core.package/ICHelper.class/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+An ICHelper is a method bucket which contains helpermethods for the filesystem storage, parameter sorting and subfolder reading.
+It ist leagcy. It should burn in hell for it sins. We should delete this.
```
Comment (EightSQ) [inline packages/IMAPClient-Core.package/ICHelper.class/README.md]: *its* sins.

File: packages/IMAPClient-Tests.package/ICDeployment.class/README.md [hunk 3]
```diff
@@ -1 +1 @@
-This class supportes the automatic release via travis.
\ No newline at end of file
+This class supportes the automatic release via travis. It is not used for teting but enables automatic sar bulding.
```
Comment (EightSQ) [inline packages/IMAPClient-Tests.package/ICDeployment.class/README.md]: This class *supports automatic releases* via Travis.
It is not used for *testing*.

File: packages/IMAPClient-Tests.package/ICEmailTest.class/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+This test the class ICEmail by testing basic formatting, header and body accessing.
```
Comment (EightSQ) [inline packages/IMAPClient-Tests.package/ICEmailTest.class/README.md]: This conducts testing of basic formatting, header and body accessing methods of the ICEmail class.

File: packages/IMAPClient-Tests.package/ICMockEndpoint.class/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+This is part of the imap server communication mock up.
```
Comment (EightSQ) [inline packages/IMAPClient-Tests.package/ICMockEndpoint.class/README.md]: I think the term *mock endpoint for testing purposes* would fit better here.

File: packages/IMAPClient-Tests.package/ICMockServer.class/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+This is part of the imap server communication mock up.
```
Comment (EightSQ) [inline packages/IMAPClient-Tests.package/ICMockServer.class/README.md]: I think the term *mock server for testing purposes* would fit better here.

File: packages/IMAPClient-UI.package/ICPasswordDialog.class/README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+An ICLoginDialog is a toolbuilder-based UI for authorizing a user. It is basically a password promt
```
Comment (EightSQ) [inline packages/IMAPClient-UI.package/ICPasswordDialog.class/README.md]: ...a password *prompt*.

File: packages/IMAPClient-Core.package/ICFetchResponseParser.class/README.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+An ICFetchResponseParser parses a IMAP-server response containing mails to a dictionary containing all these mails.
```
Comment (EightSQ) [inline packages/IMAPClient-Core.package/ICFetchResponseParser.class/README.md]: ... parses *an* IMAP server reponse to a FETCH request.

File: packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md [hunk 9]
```diff
@@ -0,0 +1 @@
+An ICFileReadWriter is a JSON File Writer and Reader to implement the file system storage of mails
```
Comment (EightSQ) [inline packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md]: I would rather describe this as *wrapper for file system storage of mails*.

**Review by EightSQ — CHANGES_REQUESTED** [review]
Sorry, but I do not like the quality of some class comments: found many typos and some clumsy wording.

File: packages/IMAPClient-Tests.package/ICMockEndpoint.class/README.md (hunk 5 above)
Comment (timgarrels) [reply packages/IMAPClient-Tests.package/ICMockEndpoint.class/README.md]: I added "offline":
"Mock endpoint for offline testing purposes"

File: packages/IMAPClient-Tests.package/ICMockServer.class/README.md (hunk 6 above)
Comment (timgarrels) [reply packages/IMAPClient-Tests.package/ICMockServer.class/README.md]: I added "offline":
"Mock server for offline testing purposes"

File: packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md (hunk 9 above)
Comment (timgarrels) [reply packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md]: I do prefer my wording as I think it describes the class better.
@Paulpanther @martintaraz @felixrindt

Comment (Paulpanther) [reply packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md]: I think Tims version is better, since it has more info

Comment (EightSQ) [reply packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md]: Okay, but please change it to *... Writer and Reader implementing the file system storage of mails.*.

Comment (EightSQ) [reply packages/IMAPClient-Core.package/ICFileReadWriter.class/README.md]: Okay, I kept @EatingBacon's version but changed the term *implement* to *handle* due to this describing what an instance of ICFileReadWriter does.

**Review by EightSQ — APPROVED** [review]

**Pull request merged**